### ISSUE TITLE
After upgrading to DSM 7, updating container users are unable to authenticate

### ISSUE THREAD

**Issue by q20** [ahaenggli/AzureAD-LDAP-wrapper#10]
Sorry, needed to use OCR to pull the error from my screen:

graph_azuread.js loginWithUsernamePassword Error: Get Token request returned http error: 400 and server response: stdout wrong passwf"error" "interaction required", "error_description": "AADSTS50076: Due to a configuration change made by your administrator, or because you moved to a new location, you must use multi-factor authentication to access '797f4846-ba00-4fd7-ba43-dac1f8f63013°. \r\nTrace ID: 9eb381ba-429d-45ce-bdc4-74f66b235500\r\Correlation ID: a32d7f75-3130-404e-a68c-9b127d9f2eed\r\nTimestamp:2021-11-18 10:00:50Z" "error codes": [50076],"timestamp":"2021-11-18 10:00:50Z", "trace _id": "9eb381ba-429d-45ce- bdc4-74f66b235500", "correlation_id": "a32d7f75-3130-404e- a68c-9b127d9f2eed" "erro

Seems security defaults would resolve this, but I am not willing to do that.
https://www.c-sharpcorner.com/article/solution-due-to-a-configuration-change-made-by-your-administrator-while-connect/

Any ideas?

**Comment by q20** [comment]
PS: I have copied over and mapped the ldap_customizer.js file (wihtout changes) and set DSM7 to true - result is unchanged... M365 users are unable to log in to DSM.

**Comment by ahaenggli** [comment]
The error says MFA (multi-factor authentication) is enabled?
If so, it has nothing to do with DSM7. MFA is just not supported by Microsoft for logins with username/password.
Some additional links are in the [FAQ](https://github.com/ahaenggli/AzureAD-LDAP-wrapper#does-it-support-mfa-multi-factor-authentication).

However, you could try activate the sso-client for the azure profile:
![image](https://user-images.githubusercontent.com/23347180/142409826-78c3f38d-f8cd-4f70-889a-4eaef1c44569.png)
maybe the usermapping works with that - but just maybe, I haven't made any tests yet.

**Comment by q20** [comment]
The users do not have MFA enabled and the accounts authenticated correctly yesterday, before updating to DSM7 and updating your container...

<img width="179" alt="image" src="https://user-images.githubusercontent.com/17024596/142410619-f0d4f38f-b24f-494b-a06e-9a5874056078.png">

The only account MFA is enabled on is our tenant admin.

**Comment by ahaenggli** [comment]
Strange. In production (docker tag "latest") I didn't change the login process. I have no idea why it doesn't work anymore... 
My NAS (with DSM 7) has no problems yet.

In development (docker tag "dev") i'm trying to switch the package azure/ms-rest-nodeauth to azure/Identity.
Therefore [Treat application as a public client](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Username-Password-Authentication#application-registration) must be set to `true`. Maybe this setting does help too in your case?

**Comment by q20** [comment]
Thanks, but I don't see that option in Authentication, unfortunately.  : (

![image](https://user-images.githubusercontent.com/17024596/142425917-e03e9122-011c-42e2-96e4-d6cc487fcb65.png)

I think I may have to give up; customer needs access ASAP and, as they're only a few people, will likely just create  users manually on the Synology. What a bummer!

**Comment by q20** [comment]
Just some additional info, in case it helps.  ; )

<img width="1147" alt="image" src="https://user-images.githubusercontent.com/17024596/142445371-521defb5-9c19-460f-b318-4d3f3760136e.png">

**Comment by ahaenggli** [comment]
The description of "Allow public client flows" sounds like the right choice, maybe ms/azure renamed the option.
However, error 400 means the request is wrong... 
Did you delete the container and made a new install? Maybe some invisible char is in the tenantId env var?

**Comment by q20** [comment]
G'day

Unfortunately I will no longer be able to troubleshoot this issue; we've reverted to local users/groups.
I can tell you, though, that we used exactly the same configuration file as before, having exported a JSON backup from the Synology docker package. The only change was an upgraded container image (and mapping the .JS file). We tried importing again without the .JS and suffered the same issue.

Anyway, I thank you for your time (again).  : )
